contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -342,17 +463,253 @@ abstract contract BaseHarvester is Governable {
+            _swap(tokenConfig, _swapToken, _rewardTo, _swapWithCurve);
+        } else {
+            revert("Unknown swap platform");
+        }
+    }
+
+    /**
+     * @dev Swap a reward token for the base token on the configured
+     *      swap platform. The token must have a registered price feed
+     *      with the price provider.
+     * @param tokenConfig RewardTokenConfig for the token to swap
+     * @param _swapToken Address of the token to swap
+     * @param swapper Pointer to the actual swap function
+     */
+    function _swap(
+        RewardTokenConfig memory tokenConfig,
+        address _swapToken,
+        address _rewardTo,
+        function(RewardTokenConfig memory, address, uint256, uint256)
+            returns (uint256) swapper
+    ) internal virtual {
+        // RewardTokenConfig memory tokenConfig = rewardTokenConfigs[_swapToken];
+
+        /* This will trigger a return when reward token configuration has not yet been set
+         * or we have temporarily disabled swapping of specific reward token via setting
+         * doSwapRewardToken to false.
+         */
+        if (!tokenConfig.doSwapRewardToken) {
+            return;
+        }
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+        uint256 balance = IERC20(_swapToken).balanceOf(address(this));
+
+        if (balance == 0) {
+            return;
+        }
+
+        if (tokenConfig.liquidationLimit > 0) {
+            balance = Math.min(balance, tokenConfig.liquidationLimit);
+        }
+
+        // This'll revert if there is no price feed
+        uint256 oraclePrice = IOracle(priceProvider).price(_swapToken);
+
+        // Oracle price is 1e18
+        uint256 minExpected = (balance *
+            (1e4 - tokenConfig.allowedSlippageBps) * // max allowed slippage
+            oraclePrice).scaleBy(
+                baseTokenDecimals,
+                Helpers.getDecimals(_swapToken)
+            ) /
+            1e4 / // fix the max slippage decimal position
+            1e18; // and oracle price decimals position
+
+        // Do the swap
+        uint256 amountReceived = swapper(
+            tokenConfig,
+            _swapToken,
+            balance,
+            minExpected
+        );
+        emit RewardTokenSwapped(
+            _swapToken,
+            baseTokenAddress,
+            tokenConfig.platform,
+            balance,
+            amountReceived
+        );
+
+        IERC20 desiredToken = IERC20(baseTokenAddress);
+        uint256 desiredTokenBalance = desiredToken.balanceOf(address(this));
+
+        uint256 vaultBps = 1e4 - tokenConfig.harvestRewardBps;
+        uint256 rewardsProceedsShare = (desiredTokenBalance * vaultBps) / 1e4;
+
+        require(
+            vaultBps > tokenConfig.harvestRewardBps,
+            "Address receiving harvest incentive is receiving more rewards than the rewards proceeds address"
+        );
+
+        uint256 farmerShare = desiredTokenBalance - rewardsProceedsShare;
```

Comment:
This share calculation is a bit convoluted, and is actually rounding against the vault, in favor of the farmer. 

I think this can much simpler if we just calculate the farmer share first, then the vault share by subtraction, then compare those two totals.